idflakies-maven-plugin/src/main/java/edu/illinois/cs/dt/tools/plugin/IncDetectorMojo.java

```diff
@@ -0,0 +1,635 @@
+            // don't compute changed and non-affected classes
+            // Make nonAffected empty so dependencies can be updated
+            Writer.writeClassPath(cpString, artifactsDir);
+            writeJarChecksums(sfPathElements, artifactsDir, jarCheckSums);
+            selectAll = true;
+        }
+    }
+    private ClassLoader createClassLoader(Classpath sfClassPath) {
+        long start = System.currentTimeMillis();
+        ClassLoader loader = null;
+        try {
+            loader = sfClassPath.createClassLoader(false, false, "MyRole");
+        } catch (SurefireExecutionException see) {
+            see.printStackTrace();
+        }
+        long end = System.currentTimeMillis();
+        Logger.getGlobal().log(Level.FINE, "[PROFILE] IncDetectorPlugin(createClassLoader): "
+                + Writer.millsToSeconds(end - start));
+        return loader;
+    }
+
+    public Path relativePath(final Path initial, final Path relative) {
+        Preconditions.checkState(!relative.isAbsolute(),
+                "PathManager.path(): Cache paths must be relative, not absolute (%s)", relative);
+
+        return initial.resolve(relative);
+    }
+
+    @Override
+    protected Void defineSettings(final ErrorLogger logger, final MavenProject project) throws IOException {
+        super.defineSettings(logger, project);
+
+        artifactsDir = getArtifactsDir();
+        startsDir = getStartsDir();
+        ekstaziDir = getEkstaziDir();
+        selectMore = Configuration.config().getProperty("dt.incdetector.selectmore", false);
+        selectAll = false;
+        detectOrNot = Configuration.config().getProperty("dt.incdetector.detectornot", true);
+        isEkstazi = Configuration.config().getProperty("dt.incdetector.ekstazi", false);
+        ekstaziSelectedTestsPath = relativePath(PathManager.ekstaziPath(), Paths.get("selected-tests"));
+        // ekstaziDependenciesFile = Configuration.config().getProperty("dt.incdetector.ekstazidependencies", "");
+        startsSelectedTestsPath = relativePath(PathManager.startsPath(), Paths.get("selected-tests"));
+        startsDependenciesPath = relativePath(PathManager.startsPath(), Paths.get("deps.zlc"));
+
+        transitiveClosure = new HashMap<>();
+        reverseTransitiveClosure = new HashMap<>();
+
+        getSureFireClassPath(project);
+        loader = createClassLoader(sureFireClassPath);
+
+        return null;
+    }
+
+    private String getArtifactsDir() throws FileNotFoundException {
+        if (artifactsDir == null) {
+            artifactsDir = PathManager.cachePath().toString();
+            File file = new File(artifactsDir);
+            if (!file.mkdirs() && !file.exists()) {
+                throw new FileNotFoundException("I could not create artifacts dir: " + artifactsDir);
+            }
+        }
+        return artifactsDir;
+    }
+
+    private String getStartsDir() throws FileNotFoundException {
```

Comment:
`getStartsDir` and `getEkstaziDir` look so similar, I wonder if the logic can be somewhat merged.